### ISSUE TITLE
Merge-NAVVersionList sorts versions as text; fails when comparing e.g. '9.00' and '10.00'

### ISSUE THREAD

**Issue by jhoek** [waldo1001/Cloud.Ready.Software.PowerShell#6]
Using the latest version of Merge-NAVVersionList from your GitHub repo, testing with the following data:

``` powershell
$MyVersionprefix = 'NAVW1', 'NAVBE', 'I'
$OrigVersionlist = 'NAVW19.10.00.37563,NAVBE7.10.00.37563'
$ModVersionList = 'NAVW19.10.00.37563,NAVBE7.10.00.37563,I7.4,M22344,M65877'
$TgtVersionList = 'NAVW110.00,NAVBE8.00'

Merge-NAVVersionList `
    -OriginalVersionList $OrigVersionlist `
    -ModifiedVersionList $ModVersionList `
    -TargetVersionList $TgtVersionList `
    -Versionprefix $MyVersionprefix
```

...will produce the following (incorrect, AFAIAC) result:

`NAVW19.10.00.37563,NAVBE8.00,I7.4,M22344,M65877`

Versions should probably not be sorted as simple text?!

**Comment by rdebath** [comment]
You should probably also note that according to the import worksheet duplicating the version tags is a "no-no". 

If you try to import an object with "M22344,M65877" over itself it will always conflict because M65877 is later than M22344 and M22344 is earlier than M65877

**Comment by waldo1001** [comment]
Thanks for this .. looks like I'm going to have to review this for the next version ;-).
Any suggestions?

Just a note on the M.. numbers: we use this as internal, temporary issue numbers.  They get removed when we release a database.  So in (all?) most upgrades, they don't exist - or they only exist in one database, which is the customer-database .. .

**Comment by rdebath** [comment]
If you want to put something into a Version List that is not a "version" the best you can do to keep the import worksheet happy is probably to attach it to the end of an existing "version"

For example these "version list"s are considered identical: 
    `NAVW18.00,M0-22344` and `NAVW18.00,M0-65877`

Both of those are identical to `NAVW18.00,M0-22344,M0-65877` and `NAVW18.00,M0`

The "Version Tag" is the piece upto (and not including) the first character from `0.123456789` the version number is the run of digits and dots. Anything after that is ignored. NB: only the first run of digits are compared using the numeric value, digits after the first `.` are compared as a string.

Oh and while I'm on about this, the "Version tag" is compared as a Nav code field, so it's case insensitive AND both leading and trailing spaces are trimmed.

So in summary ...
 `NAVW18.00,ABC1.23` is the same as `NAVW18.00,   aBc  001.23/Don't release this version`

**Comment by jhoek** [comment]
> Any suggestions?

Wel... maybe. What I'm working on right now is a solution where I convert each of the elements (NAVW19.10.00.37563, MyApp1.00 etc.) into custom PSObjects that have the following properties:
- Prefix (e.g. NAVW1)
- VersionText (e.g. 9.10.00.37563)
- VersionNo (e.g. 9.100037563)

The VersionNo property is the same as VersionText, but with all-but-the-first full stop removed, and converted to a decimal. That, at least, gives me a meaningful value to use in comparisons, sorting etc. 

Still testing my code, but the results look promising so far... ;-)

**Comment by jhoek** [comment]
Oops. Wait! No! Not closed! Clumsy me... ;-)

**Comment by jhoek** [comment]
What complicated my code the most is, ironically, the NAVW1 version element prefix. Who in their right mind would dream up a prefix that ends with a digit?!... ;-)

**Comment by jhoek** [comment]
@waldo1001 With NAV2017 (= NAVW110.0?) released, this has just become a bit more urgent, hasn't it?

**Comment by waldo1001** [comment]
@jhoek 
Thanks for the heads up.  I think I have a solution.  Just checked in.  It would be great if you can test it :-).

This is the commit that should fix it:
https://github.com/waldo1001/Cloud.Ready.Software.PowerShell/commit/1826879487ab0e277923959542f6d0f287b2c3d5
